### DISCUSSION TITLE
Is there a way to do stable random search?

### DISCUSSION THREAD

**Discussion by lanyusan** [marqo-ai/marqo#463] [Q&A] [answered]
My use case is that I wish to randomly present a feed to each refresh by user for items in a certain collection, for example photos.

Yet when user scroll down, this random generated feed should be stable and have no duplicates or few duplicates.

Each refresh could generate a random seed, then this seed can be used for this search.

I am not sure if it is an appropriate application of vector search. Any insight would be much appreciated.

Thanks.

**Comment by jn2clark** [comment, marked as answer]
Hi @lanyusan, there are a couple of ways. If you know the `_id` list ahead of time you could randomly sample a document and then use that to search with. Alternatively, we have a `context_vector` that can be used for a random search. In this case, you would create a random vector and then search directly with that. Here is an example;
```python
res = client.search({query:0.0}, context={"tensor":[{"vector": context_vector, "weight": 1.0}]})
```
where `context_vector` is a random vector.

**Reply by aryanverma22** [reply]
Hey @jn2clark , is there any documentation available about how these vectors are used to search in our Index and what their range is, in order to be able to generate them randomly. Right now I am getting similar search results for a huge range of these vectors, using the method you have mentioned above.
